XamlControlsGallery/PageHeader.xaml

```diff
@@ -114,6 +115,38 @@
                         <AppBarButton.Icon>
                             <FontIcon Glyph="&#xE771;" Margin="0,-2,0,0" />
                         </AppBarButton.Icon>
+                        <AppBarButton.Resources>
+                            <controls:TeachingTip x:Name="ToggleThemeTeachingTip1"                   
+                            Target="{x:Bind ThemeButton}"
+                            Title="Change themes without hassle"                                        
+                            Subtitle="It's easier than ever to see control samples in both light and dark theme!">
+                                <controls:TeachingTip.IconSource>
+                                    <controls:SymbolIconSource Symbol="Refresh" />
+                                </controls:TeachingTip.IconSource>
+                            </controls:TeachingTip>
+                            <controls:TeachingTip x:Name="ToggleThemeTeachingTip2"                   
```

Comment:
It feels weird to me to put the tip sample code outside of the tip page. I know you can't use x:Bind to an element name but you could have a property of the TeachingTipPage itself that grabs the one out of the frame.

Reply:
There were a few reasons we did this:

- We could recreate sample UI to point to in the sample page, but it violates our guidance to not put targeted non-light-dismiss tips on scrollable UI, it adds visual complexity to the sample, and using teaching tips in an a real scenario for the gallery felt more in the spirit of the control than reconstructing a copy of the "Toggle Theme" button (or contrived scenario) on the page when the UI already exists.
- Putting the teaching tips here let them easily access their target, the navigation events to close on, and one teaching tip hosts a "Toggle theme now" button that otherwise need to get a handle into this class. Passing all of this required a few globals and weird hacks that Steven preferred to avoid.
- If the teaching tips were inside the sample page, the effects of the "Toggle Theme" button, as implemented in the gallery, would cascade poorly onto visible teaching tips. We were not sure why. I believe it had to do with the way this is implemented in the gallery as open teaching tips respect system-wide theme changes perfectly.

Reply:
With all that noted, it appears if the theme change issue was a bug on TeachingTip, then we could migrate these samples back to the sample page if Steven is OK setting up globals for the target and events the samples would need.

Reply:
Yes, the theme change bug is a product issue. I've got a fix out for PR: Microsoft/microsoft-ui-xaml#488.

Reply:
@stmoy With the theme change issue out of the way, would you prefer me to migrate the tips to their sample page and use static handles to the target, navigation events, and toggle theme method?

Reply:
I agree with Jevan - the sample code should live on the sample page if possible.

Reply:
Turns out this is not currently achievable as the Gallery does not properly surface navigation events to Sample pages. A bug will be filed for this issue and I'll keep the teaching tips in the header file until it is resolved.